libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2211,7 +2229,12 @@ string YulUtilFunctions::readFromStorageValueType(Type const& _type, optional<si
 
 string YulUtilFunctions::readFromStorageReferenceType(Type const& _type)
 {
-	solUnimplementedAssert(_type.category() == Type::Category::Struct, "");
+	if (auto const* arrayType = dynamic_cast<ArrayType const*>(&_type))
+		return copyArrayFromStorageToMemoryFunction(
+			dynamic_cast<ArrayType const&>(*arrayType->copyForLocation(DataLocation::Storage, false)),
+			*arrayType
```

Comment:
Why do you know that arrayType is in memory? Shouldn't it rather be in storage?

Reply:
This should be type to what we are trying to read, which is a bit misleading. I should:
1. rename function to have better name, as this is basically copying from storage to memory for reference types
2. add additional argument to have `_fromType` and `_toType`

Sounds fine?

Reply:
If it works then it's fine.

Reply:
But maybe add an assertion here to make it explicit? I know there is an assertion in the called function, but still.

Reply:
Added additional assertion